api_gateway/src/main/kotlin/pl/edu/pg/rsww/apigateway/RoutingController.kt

```diff
@@ -27,6 +25,36 @@ public class RoutingController(
         return "Hello, I'm alive!"
     }
 
+    @RequestMapping("/api/mocked/{*path}")
+    fun mock(
+        @PathVariable path: String,
+        @RequestParam params: Map<String, String>,
+    ): ResponseEntity<String> {
+        val builder = ResponseEntity.status(200)
+        if (path.contains("get_trips")) {
+            val n: Int = params["n"]?.toInt() ?: 0
+            val next = n + 1
+
+            return builder.body(
+                """{
+	  "name": "${params["search"]}",
+	  "n": $next,
+	  "id": $n,
+	  "description": "Hello, world!"
+      }""",
+            )
+        }
+        if (path.contains("trip")) {
+            return builder.body(
+                """{
+	  "name": "Trip no. ${params["id"]}",
+	  "description": "A slightly longer description for trip number ${params["id"]}"
+      }""",
+            )
+        }
+        return builder.body("""{ "error": "404" }""")
```

Comment:
No reason, but since error codes are not handled anyway on the front-end there is no reason to return any other status in this mocked endpoint. Even more there are no transport layer error reported in this snipped to justify returning non 2xx error code.